### PR TITLE
Create Sort_two_array_without_extra_space.md

### LINKED ISSUE #7069: Two sorted Arrays without extra space
### Description

Given two sorted arrays, we need to merge them in O((n+m)*log(n+m)) time with O(1) extra space into a sorted array, when n is the size of the first array, and m is the size of the second array.

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by vineetsingh0809**
<hr>

## Description 📜
Added a problem in which given two sorted arrays, we need to merge them in O((n+m)*log(n+m)) time with O(1) extra space into a sorted array, when n is the size of the first array, and m is the size of the second array. 
Fixes #7069 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [x] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: C_CPP/Competitive Programming/Sort_two_array_without_extra_space.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Two sorted Arrays without extra space
+
+## Problem Statement
+
+Given two sorted arrays, we need to merge them in O((n+m)\*log(n+m)) time with O(1) extra space into a sorted array, when n is the size of the first array, and m is the size of the second array.
+
+### EXAMPLE
```
Comment (deeqakkk) [inline C_CPP/Competitive Programming/Sort_two_array_without_extra_space.md:7]: use a borderless table to align them properly.

File: C_CPP/Competitive Programming/Sort_two_array_without_extra_space.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ar2[] = {25, 34}
+
+Input: ar1[] = {1, 5, 9, 10, 15, 20};
+ar2[] = {2, 3, 8, 13};
+Output: ar1[] = {1, 2, 3, 5, 8, 9}
+ar2[] = {10, 13, 15, 20}
+
+### ALGORITHM
+
+We start comparing elements that are far from each other rather than adjacent.
```
Comment (apeksha235) [inline C_CPP/Competitive Programming/Sort_two_array_without_extra_space.md:21]: This is a shell sort algorithm, please mention it

File: C_CPP/Competitive Programming/Sort_two_array_without_extra_space.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+For every pass, we calculate the gap and compare the elements towards the right of the gap. Every pass, the gap reduces to the ceiling value of dividing by 2.
+
+### C++ Programm code of the Problem
```
Comment (deeqakkk) [inline C_CPP/Competitive Programming/Sort_two_array_without_extra_space.md:24]: fix this typo as well "Programm"

File: C_CPP/Competitive Programming/Sort_two_array_without_extra_space.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+```
+First Array - { 3, 9, 10, 27, 28 }
+Second Array - { 43, 82 }
+```
+
+**COMPLEXITY**
+
+```
+Time Complexity - O(n log n)
```
Comment (apeksha235) [inline C_CPP/Competitive Programming/Sort_two_array_without_extra_space.md:106]: explain the time complexity in your own words

**Review by deeqakkk — CHANGES_REQUESTED** [review]
image
If you have used content from other websites do add their link as a referance.

**Review by apeksha235 — CHANGES_REQUESTED** [review]
@vineetsingh0809 make these changes, also explain the algorithm in points for better clarity

**Comment by vineetsingh0809** [comment]
Hey @deeqakkk and @apeksha235, changes are done, now review it again.

File: C_CPP/Competitive Programming/Sort_two_array_without_extra_space.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
+# Two sorted Arrays without extra space
+
+## Problem Statement
+
+Given two sorted arrays, we need to merge them in O((n+m)\*log(n+m)) time with O(1) extra space into a sorted array, when n is the size of the first array, and m is the size of the second array.
+
+### EXAMPLE
+
+Input: ar1[] = {25};
```
Comment (apeksha235) [inline C_CPP/Competitive Programming/Sort_two_array_without_extra_space.md:9]: add the input and output inside ``` for better presentation

**Review by apeksha235 — COMMENTED** [review]
@vineetsingh0809 it seems like you've taken the code entirely from [here](https://www.geeksforgeeks.org/efficiently-merging-two-sorted-arrays-with-o1-extra-space/) 
you can definitely refer from different websites but taking the code entirely is not permitted. 
@Suvraneel

**Comment by Suvraneel** [comment]
@vineetsingh0809  On second thought.... I'll let this be open... Use your own words... remove plagiarism

**Pull request closed without merging**